File: app/src/main/java/ru/strcss/projects/moneycalc/moneycalcandroid/componets/spendingsections/addeditspendingsection/AddEditSpendingSectionActivity.java
package ru.strcss.projects.moneycalc.moneycalcandroid.componets.spendingsections.addeditspendingsection;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatDelegate;
import android.support.v7.widget.Toolbar;

import javax.inject.Inject;

import dagger.Lazy;
import dagger.android.support.DaggerAppCompatActivity;
import moneycalcandroid.moneycalc.projects.strcss.ru.moneycalc.R;
import ru.strcss.projects.moneycalc.enitities.SpendingSection;
import ru.strcss.projects.moneycalc.moneycalcandroid.utils.ActivityUtils;

import static ru.strcss.projects.moneycalc.moneycalcandroid.AppConstants.SPENDING_SECTION;

public class AddEditSpendingSectionActivity extends DaggerAppCompatActivity {

    @Inject
    AddEditSpendingSectionPresenter presenter;

    @Inject
    Lazy<AddEditSpendingSectionFragment> addEditSpendingSectionFragmentProvider;

    private ActionBar mActionBar;

    static {
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.addeditspendingsection_activity);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        mActionBar = getSupportActionBar();
        mActionBar.setDisplayHomeAsUpEnabled(true);
        mActionBar.setDisplayShowHomeEnabled(true);

        // TODO: 01.05.2018 insert transactionId here
//        getIntent().getAction()
        SpendingSection editedSpendingSection = null;
        if (getIntent().getExtras() != null) {
            editedSpendingSection = (SpendingSection) getIntent().getExtras().getSerializable(SPENDING_SECTION);
        }
        setToolbarTitle(editedSpendingSection);

        AddEditSpendingSectionFragment addEditSpendingSectionFragment =
                (AddEditSpendingSectionFragment) getSupportFragmentManager().findFragmentById(R.id.addEditSpendingSection_contentFrame);
        if (addEditSpendingSectionFragment == null) {
            // Get the fragment from dagger
            addEditSpendingSectionFragment = addEditSpendingSectionFragmentProvider.get();

            Bundle bundle = new Bundle();
            bundle.putSerializable(SPENDING_SECTION, editedSpendingSection);
            addEditSpendingSectionFragment.setArguments(bundle);

            ActivityUtils.addFragmentToActivity(
                    getSupportFragmentManager(), addEditSpendingSectionFragment, R.id.addEditSpendingSection_contentFrame);
        }
    }

    private void setToolbarTitle(@Nullable SpendingSection spendingSection) {
        if (spendingSection == null) {
            mActionBar.setTitle(R.string.spending_section_add);
        } else {
            mActionBar.setTitle(R.string.spending_section_edit);
        }
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }
}

File: app/src/main/java/ru/strcss/projects/moneycalc/moneycalcandroid/componets/spendingsections/addeditspendingsection/AddEditSpendingSectionActivity.kt
package ru.strcss.projects.moneycalc.moneycalcandroid.componets.spendingsections.addeditspendingsection

import android.os.Bundle
import android.support.v7.app.ActionBar
import android.support.v7.app.AppCompatDelegate
import android.support.v7.widget.Toolbar
import dagger.Lazy
import dagger.android.support.DaggerAppCompatActivity
import moneycalcandroid.moneycalc.projects.strcss.ru.moneycalc.R
import ru.strcss.projects.moneycalc.moneycalcandroid.AppConstants.SPENDING_SECTION
import ru.strcss.projects.moneycalc.moneycalcandroid.utils.ActivityUtils
import ru.strcss.projects.moneycalc.moneycalcdto.entities.SpendingSection
import javax.inject.Inject

class AddEditSpendingSectionActivity : DaggerAppCompatActivity() {

    @Inject
    lateinit var presenter: AddEditSpendingSectionPresenter

    @Inject
    lateinit var addEditSpendingSectionFragmentProvider: Lazy<AddEditSpendingSectionFragment>

    private var mActionBar: ActionBar? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.addeditspendingsection_activity)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        mActionBar = supportActionBar
        mActionBar!!.setDisplayHomeAsUpEnabled(true)
        mActionBar!!.setDisplayShowHomeEnabled(true)

        // TODO: 01.05.2018 insert transactionId here
        //        getIntent().getAction()
        var editedSpendingSection: SpendingSection? = null
        if (intent.extras != null) {
            editedSpendingSection = intent.extras!!.getSerializable(SPENDING_SECTION) as SpendingSection
        }
        setToolbarTitle(editedSpendingSection)

        var addEditSpendingSectionFragment: AddEditSpendingSectionFragment? =
                supportFragmentManager.findFragmentById(R.id.addEditSpendingSection_contentFrame) as? AddEditSpendingSectionFragment?
        if (addEditSpendingSectionFragment == null) {
            // Get the fragment from dagger
            addEditSpendingSectionFragment = addEditSpendingSectionFragmentProvider.get()

            val bundle = Bundle()
            bundle.putSerializable(SPENDING_SECTION, editedSpendingSection)
            addEditSpendingSectionFragment!!.arguments = bundle

            ActivityUtils.addFragmentToActivity(
                    supportFragmentManager, addEditSpendingSectionFragment, R.id.addEditSpendingSection_contentFrame)
        }
    }

    private fun setToolbarTitle(spendingSection: SpendingSection?) {
        if (spendingSection == null) {
            mActionBar!!.setTitle(R.string.spending_section_add)
        } else {
            mActionBar!!.setTitle(R.string.spending_section_edit)
        }
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }

    companion object {

        init {
            AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
        }
    }
}
